### PR TITLE
crio wipe: ensure a clean shutdown

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
there are cases where crio doesn't get the chance to sync before shutdown.
In these cases, container storage can be corrupted.
We need to protect against this case by wiping all of storage if we detect we didn't cleanly shutdown.

Add an option to specify a clean_shutdown_file that crio will create upon syncing at shutdown
Add an option to crio-wipe to clear all of storage if that file is not present
Add integration tests to verify

#### Does this PR introduce a user-facing change?

```release-note
add clean_shutdown_file option to allow crio/crio wipe to verify crio had time to shutdown cleanly
```

### CODE REVIEW

File: docs/crio.8.md [hunk 1]
```diff
@@ -117,6 +118,8 @@ crio [GLOBAL OPTIONS] command [COMMAND OPTIONS] [ARGUMENTS...]
 
 **--cgroup-manager**="": cgroup manager (cgroupfs or systemd) (default: systemd)
 
+**--clean-shutdown-file**="": Location for CRI-O to lay down the clean shutdown file. It indicates whether we've had time to sync changes to disk before shutting down. If not found, crio wipe will clear the storage directory (default: /var/lib/crio/clean.shutdown)
+
```
Comment (TomSweeneyRedHat) [inline docs/crio.8.md:122]: :+1:

File: internal/criocli/criocli.go [hunk 2]
```diff
@@ -777,6 +780,13 @@ func getCrioFlags(defConf *libconfig.Config) []cli.Flag {
 			EnvVars:   []string{"CONTAINER_VERSION_FILE_PERSIST"},
 			TakesFile: true,
 		},
+		&cli.StringFlag{
+			Name:      "clean-shutdown-file",
+			Usage:     "Location for CRI-O to lay down the clean shutdown file. It indicates whether we've had time to sync changes to disk before shutting down. If not found, crio wipe will clear the storage directory",
```
Comment (TomSweeneyRedHat) [inline internal/criocli/criocli.go:785]: The verbiage is awesome, but is it too long for the usage?  I'd a vague memory that we wanted (required?) these to be under 80 chars.

**Comment by TomSweeneyRedHat** [comment]
LGTM, one question

**Comment by umohnani8** [comment]
/retest
LGTM

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by giuseppe** [comment]
@mrunalp @umohnani8 @saschagrunert PTAL

**Review by saschagrunert — APPROVED** [review]
LGTM

**Comment by giuseppe** [comment]
@haircommander a merge conflict :(

**Comment by haircommander** [comment]
ref bz https://bugzilla.redhat.com/show_bug.cgi?id=1942536

**Pull request merged**